app/models/user.rb

```diff
@@ -27,6 +27,7 @@ class User < ApplicationRecord
   before_save :geocode, if: ->(v) { v.zip.present? && v.zip_changed? }
   before_save :upcase_state
   before_save :downcase_email
+  before_save :validate_zipcode
```

Comment:
Instead of a callback, can we convert this into a [custom validation method](http://guides.rubyonrails.org/active_record_validations.html#custom-methods).